core/app/models/spree/taxon.rb

```diff
@@ -66,10 +67,12 @@ class Taxon < Spree::Base
 
       def set_permalink
         parent = translated_model.parent
+        name_with_fallback = name || translated_model.name
```

Comment:
Just a question - this refers to the `name` in the translation and when it's not set it falls back to the model's `name` which kicks in the whole fallback mechanism, right?

Reply:
Exactly